addons/stock/models/product_strategy.py

```diff
@@ -80,11 +80,13 @@ def _domain_product_id(self):
     @api.onchange('location_in_id')
     def _onchange_location_in(self):
         child_location_count = 0
-        if self.location_out_id:
+        location_out = self.location_out_id._origin
+        location_in = self.location_in_id._origin
+        if location_out and location_in:
             child_location_count = self.env['stock.location'].search_count([
-                ('id', '=', self.location_out_id.id),
-                ('id', 'child_of', self.location_in_id.id),
-                ('id', '!=', self.location_in_id.id),
+                ('id', '=', location_out.id),
+                ('id', 'child_of', location_in.id),
+                ('id', '!=', location_in.id),
```

Comment:
This search_count is in fact a basic check on the relation between the two locations.
why not changing it by 
```python
if loc_in.child_of(loc_out) and loc_in != loc_out:
   # reset loc_out

Reply:
@nle-odoo That is a very good point ! I was convinced that both expression were equivalent last time I checked. I thought that path always started by `/` and hence that the issue could not happen because `/1/` is not in `/11/` but if paths do not start with `/` it should indeed be changed !!